### PR TITLE
Feature/swap and next blocks

### PR DESCRIPTION
**Pull request by Nitepone**
Adds support into the Tetris library for #12

### CODE REVIEW

File: src/tetris_game.c [hunk 1]
```diff
@@ -41,32 +42,54 @@ int destroy_block(struct active_block **block) {
...
 /*
- * Generates a random block
+ * Generates a block at the top of the game board
  */
-int generate_block(struct active_block **block) {
+int spawn_active_block(struct active_block **block, enum block_type type) {
 	// allocate and fill
-	(*block)->block_type = rand() % (BLOCK_TYPE_COUNT);
+	(*block)->block_type = type;
+	//(*block)->block_type = rand() % (BLOCK_TYPE_COUNT);
```
Comment (bitoffdev) [inline src/tetris_game.c]: Probably better to just get rid of this comment

File: src/tetris_game.h [hunk 2]
```diff
@@ -103,6 +112,13 @@ int rotate_block(int clockwise, struct game_contents *game_contents);
  */
 int new_game(struct game_contents **game_contents);
 
+/*
+ * Starts a game and takes a double pointer for game contents data.
+ * Also takes a seed to base the number generation from.
+ * @return 0
+ */
+int new_seeded_game(struct game_contents **game_contents);
```
Comment (bitoffdev) [inline src/tetris_game.h]: It seems like this function signature doesn't match the one present in `src/tetris_game.c`

**Comment by bitoffdev** [comment]
It would also be good to update the formatting with `make lint` or equivalently `clang-format --style=file -i *.c *.h`. We should probably document this somewhere, maybe in  #10.

I opened another issue (#13) to add this to CI too so that the clang-format becomes a requirement for future PRs.

**Comment by bitoffdev** [comment]
Other than the small comments I left, it looks great! I cloned it and ran it successfully.

**Review by bitoffdev — CHANGES_REQUESTED** [review]
Other than the small comments I left, it looks great! I cloned it and ran it successfully.

File: src/tetris_game.h (hunk 2 above)
Comment (Nitepone) [reply src/tetris_game.h]: Do not know how this got through compilation for me..

Comment (bitoffdev) [reply src/tetris_game.h]: You have a better compiler than I do :smile:

**Review by bitoffdev — APPROVED** [review]

**Pull request merged**